source/base/partitioner.cc

```diff
@@ -174,7 +174,8 @@ namespace Utilities
       n_ghost_indices_data = ghost_indices_data.n_elements();
 
       have_ghost_indices =
-        Utilities::MPI::sum(n_ghost_indices_data, communicator) > 0;
+        Utilities::MPI::sum((n_ghost_indices_data > 0) ? 1 : 0, communicator) >
+        0;
```

Comment:
You're going to run into the same issue once we have 4B processes. That's going to be awkward to debug :-)

Why not just use `MPI::max` with the original data type? That's guaranteed to work.

Reply:
> You're going to run into the same issue once we have 4B processes. That's going to be awkward to debug :-)

I hate to tell you, but ranks are identified using a 32bit ``int`` everywhere inside MPI. ;-)

Reply:
Oh, gosh darn! It's like "Nobody needs more than 640 kB of memory over again!"